projects/103_hw_megan/src/main.c

```diff
@@ -0,0 +1,64 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+// logs the output of the ADC reading from pin A6
+static void prv_log_adc_reading(const GpioAddress *address, void *context) {
+  GpioAddress *adc_addr = context;
+  uint16_t adc_data = 0;
+  adc_read_converted_pin(*adc_addr, &adc_data);
+
+  LOG_DEBUG("The ADC converted reading is: %d\n", adc_data);
+}
+
+int main() {
+  interrupt_init();
+  gpio_init();
+
+  // gpio pin 6
+  GpioAddress adc_addr = {
+    .port = GPIO_PORT_A,
+    .pin = 6,
+  };
+
+  GpioSettings adc_settings = {
```

Comment:
We should keep the way we initialize these structs consistently.
It is best to explicitly state which setting are charging (similar to what you have in adc_addr)

```
eg
.direction = GPIO_DIR_IN
.state = ...
.resistor = ...
```